main/src/main/java/org/mobilitydata/gtfsvalidator/validator/RouteNameValidator.java

```diff
@@ -62,6 +74,22 @@ public void validate(GtfsRoute entity, NoticeContainer noticeContainer) {
           new RouteShortNameTooLongNotice(
               entity.routeId(), entity.csvRowNumber(), entity.routeShortName()));
     }
+
+    // check if route_long_name begins with route_short_name followed by " ", "-", or "(".
+    // as referenced here
+    // https://github.com/MobilityData/gtfs-validator/pull/501#discussion_r535506016
+    if (hasLongName && hasShortName) {
+      String longName = entity.routeLongName();
+      String shortName = entity.routeShortName();
+      Pattern pattern = Pattern.compile("^" + shortName + "[\\s\\-\\(]");
+      Matcher matcher = pattern.matcher(longName);
```

Comment:
I don't think you are using any specific features of Matcher (e.g. capture groups), so I'd just use String.matches() directly.